packages/vite/src/vite.ts

```diff
@@ -138,13 +138,23 @@ export const bundle: NuxtBuilder['bundle'] = async (nuxt) => {
     ctx.config.build!.watch = undefined
   }
 
-  // Trigger vite to optimize any dependencies that are imported within a layer, just as if they were imported by a user
-  if (nuxt.options.dev) {
```

Comment:
Not only in dev anymore?

Reply:
ah. yes. probably best to keep in dev for now until tested better. we might later expand out of dev to try and resolve dependencies from final layer (would resolve https://github.com/nuxt/nuxt/issues/25747, for example)...